src/main/java/test/project/bookshop/service/impl/BookServiceImpl.java

```diff
@@ -1,28 +1,36 @@
 package test.project.bookshop.service.impl;
 
 import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
-import test.project.bookshop.model.Book;
+import test.project.bookshop.dto.BookDto;
+import test.project.bookshop.dto.BookRequestDto;
+import test.project.bookshop.exception.EntityNotFoundException;
+import test.project.bookshop.mapper.BookMapper;
 import test.project.bookshop.repository.BookRepository;
 import test.project.bookshop.service.BookService;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(BookRequestDto bookRequestDto) {
+        return bookMapper.toDto(bookRepository.save(bookMapper.toBook(bookRequestDto)));
```

Comment:
bad practice to call several methods inline

```suggestion
        Book book = bookMapper.toBook(bookRequestDto);
        return bookMapper.toDto(bookRepository.save(book));
```